build-tools/md/main.go

```diff
@@ -170,6 +185,26 @@ func updateLinks(n node) {
 	mdl.URL = u.String()
 }
 
+// updateHeading inserts a [markdown.Inline] element at the start of a
+// [markdown.Heading]s Text slice that renders a link marker linking
+// to the heading, allowing for easily copying links to a heading.
+func (w *walker) updateHeading(h *markdown.Heading) {
+	text := inlineText(h.Text.Inline)
+	id := makeID(text, w.anchorIDs)
+	h.Text.Inline = slices.Insert(h.Text.Inline, 0, newAnchor(id))
+}
+
+func newAnchor(id string) markdown.Inline {
```

Comment:
I understand why you made this return `markdown.Inline` but an alternative would be to still return `*markdown.HTMLTag` and cast it above:
```
h.Text.Inline = slices.Insert(h.Text.Inline, 0, markdown.Inline(newAnchor(id)))
```

That is a little ugly but I do question `newAnchor` having knowledge of how the return type is being used. What if you wanted to add HTMLTag to something else that did not want an `Inline`?

What about:
```
text := inlineText(h.Text.Inline)
id := makeID(text, w.anchorIDs)
anchor := markdown.Inline(newAnchor(id))
h.Text.Inline = slices.Insert(h.Text.Inline, 0, anchor)
```